src/nvim/os_unix.c

```diff
@@ -141,7 +141,8 @@ void mch_exit(int r) FUNC_ATTR_NORETURN
   ui_flush();
   ml_close_all(true);           // remove all memfiles
 
-  event_teardown();
+  // Exit with error if main_loop did not teardown gracefully.
+  r = event_teardown() ? r : (r || 1);
```

Comment:
@oni-link You're right, that needs to be a ternary. The intention is to preserve the value of `r > 0`.